frame/assets/src/functions.rs

```diff
@@ -467,12 +473,14 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			return Ok(amount)
 		}
 
+		let details = Asset::<T, I>::get(id).ok_or(Error::<T, I>::Unknown)?;
+		ensure!(details.status == AssetStatus::Live, Error::<T, I>::AssetNotLive);
```

Comment:
this could have it's own `ensure_live(details)?;` function as it comes up a lot.

Reply:
You're right, but this logic is in a lot of places, and I don't want to add more things to this PR. I would rather as a different task